### ISSUE TITLE
obo download does not work for the files newly uploaded

### ISSUE THREAD

**Issue by marieALaporte** [bioversity/Crop-Ontology#57]
Here an example of the bug. Seems like the file name is not parsed correctly

http://www.cropontology.org/ontology/CO_020/FAO-Bioversity%20Multi-Crop%20Passport%20Descriptors

**Comment by gubi** [comment]
Just for clarification, for the [routes rule](https://github.com/bioversity/Crop-Ontology/blob/8dabd03dc08e1849015faba3652017cf32692273/main.js#L1257) the Ontology name in the URL seems not relevant because not parsed.
If you try with another ontology the download start (for eg. http://www.cropontology.org/obo/CO_325).
Please can you share me the `CO_020` source file in order to inspect logs locally?

**Comment by gubi** [comment]
Just a reminder for me:

### The event log
```
2018-08-23 08:28:04.320 PDT
/obo/CO_020
org.mozilla.javascript.EvaluatorException: Java class "[Ljava.lang.Object;" has no public instance field or method named "split". (rdf.js#10)
	at org.mozilla.javascript.DefaultErrorReporter.runtimeError(DefaultErrorReporter.java:109)
	at org.mozilla.javascript.Context.reportRuntimeError(Context.java:945)
	at org.mozilla.javascript.Context.reportRuntimeError(Context.java:1001)
	at org.mozilla.javascript.Context.reportRuntimeError2(Context.java:971)
	at org.mozilla.javascript.NativeJavaArray.get(NativeJavaArray.java:103)
	at org.mozilla.javascript.ScriptableObject.getProperty(ScriptableObject.java:2081)
	at org.mozilla.javascript.ScriptRuntime.getPropFunctionAndThisHelper(ScriptRuntime.java:2340)
	at org.mozilla.javascript.ScriptRuntime.getPropFunctionAndThis(ScriptRuntime.java:2327)
	at org.mozilla.javascript.gen.rdf_js_17._c_anonymous_2(rdf.js:10)
	at org.mozilla.javascript.gen.rdf_js_17.call(rdf.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call1(OptRuntime.java:66)
	at org.mozilla.javascript.gen.main_js_1._c_anonymous_57(main.js:1431)
	at org.mozilla.javascript.gen.main_js_1.call(main.js)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2521)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:300)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:129)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:76)
	at org.mozilla.javascript.gen.select_js_6._c_anonymous_9(select.js:152)
	at org.mozilla.javascript.gen.select_js_6.call(select.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call1(OptRuntime.java:66)
	at org.mozilla.javascript.gen.main_js_1._c_anonymous_55(main.js:1287)
	at org.mozilla.javascript.gen.main_js_1.call(main.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:86)
	at org.mozilla.javascript.gen.apejs_js_2._c_anonymous_1(apejs.js:18)
	at org.mozilla.javascript.gen.apejs_js_2.call(apejs.js)
	at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:426)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3178)
	at org.mozilla.javascript.gen.apejs_js_2.call(apejs.js)
	at apejs.ApeServlet.service(ApeServlet.java:58)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:48)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:257)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:76)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:146)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchServletRequest(JavaRuntime.java:693)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchRequest(JavaRuntime.java:655)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:625)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:455)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:462)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:320)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:321)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:313)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:459)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:274)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by gubi** [comment]
Tried on a test dev server with Java 8 and everything works fine.
So the file is no corrupted and the download works with Java 8.

In the Production Datastore this Ontology was indexed on 2017 March, 09.
Perhaps the bug is related to a bad file indexing.

![image](https://user-images.githubusercontent.com/746559/45289811-61279400-b4ee-11e8-9a22-332f8941993a.png)

**Comment by gubi** [comment]
As [here](https://github.com/bioversity/Crop-Ontology/blob/82863a9862d63664f68f054b2487cf94a4ddde22/main.js#L1903-L1910) file upload is pointed directly to Datastore `_/ah/upload/`, so filter data during inserting is not necessary.

Fixed as ba06f3adbd350c846052b5aae797844fb6d84bee
Please, reopen this issue if the problem will persist after the next release
